src/utils/SocketClient.hxx

```diff
@@ -46,6 +46,19 @@
 #include "utils/Atomic.hxx"
 #include "utils/format_utils.hxx"
 
+/** Connection status that can be sent back to the "owner" of the socket so
+ * it can update display or status information while the connection attempts
+ * are progressing.
+ */
+enum class SocketStatus
+{
+    MDNS_LOOKUP,
+    MDNS_CONNECT,
+    STATIC_CONNECT,
+    CONNECTED,
```

Comment:
I don't see this "CONNECTED" state posted anywhere.  Did I miss something?

Reply:
i think that gets implicit when the connection callback is invoked. It could probably be removed or we can add a (redundant) status update call, just before the connection callback is invoked.